[Lesson 17_Homework 17] :

TatyanaDovzhich opened the pull request:
- Create method “Navigate to Profile Page via header My Profile button”
- Added pages/profile_page.py & constants/profile_pages.py
- Added tests for profile page (verify Username, Sign Out via Sign In) with related constants and pages
- Added test for hello page (Sign Out via Sign In) with related constants and pages

TatyanaDovzhich committed:
[Lesson 17_Homework 17] :
- Create method “Navigate to Profile Page via header My Profile button”
- Added pages/profile_page.py & constants/profile_pages.py
- Added tests for profile page (verify Username, Sign Out via Sign In) with related constants and pages
- Added test for hello page (Sign Out via Sign In) with related constants and pages

TatyanaDovzhich committed:
[Lesson 17_Homework 17] : Fix#01:
- Fixed method' name "verify_sign_up_message"
instead of incorrect name "verify_sign_in_message"

QALightLecorPython commented on tests/test_profile_page.py:
```diff
@@ -0,0 +1,65 @@
+        # Verify profile username
+        profile_page.verify_profile_page_message(username=random_login.username_value)
+
+    def sign_out_from_profile_page_after_sign_in(self, start_page, random_login):
```
Test name should start with word 'test_'

TatyanaDovzhich replied:
Fixed: test  was renamed into "test_navigate_to_profile_page_via_sign_in"

QALightLecorPython commented on tests/test_profile_page.py:
```diff
@@ -0,0 +1,65 @@
+
+        # Navigate to profile page
+        profile_page = hello_page.navigate_to_profile_page()
+        return profile_page
```
1. You cannot do return from test
2. You cannot return twice

TatyanaDovzhich replied:
Fixed: strings 35 and 61 "return profile_page" was deleted

QALightLecorPython commented on tests/test_profile_page.py:
```diff
@@ -0,0 +1,65 @@
+        return profile_page
+
+        # Verify profile username
+        profile_page.verify_profile_page_message(username=random_login.username_value)
```
Why do you do verifications under the fixture ?

TatyanaDovzhich replied:
![image](https://user-images.githubusercontent.com/47011239/215155213-1e184cec-d5ef-4350-9cbb-6b1b19d25d46.png)
I check the username message on the page https://qa-complexapp.onrender.com/profile/

QALightLecorPython commented on tests/test_hello_page.py:
```diff
@@ -0,0 +1,32 @@
+        hello_page.verify_sign_up_message(username=random_login.username_value)
+
+        # Click on "Sign Out" button
+        header.sign_out_button()
```
What is header ? From where did you get it ?

TatyanaDovzhich replied:
Fixed: reference to hello_page was added into Click on "Sign Out" button & Click on "My Profile" button Steps (tests/test_hello_page.py & tests/test_profile_page.py

QALightLecorPython commented on tests/test_hello_page.py:
```diff
@@ -0,0 +1,32 @@
+
+    log = logging.getLogger("[TestHelloPage]")
+
+    def sign_out_from_hello_page_after_sign_in(self, start_page, random_login):
```
Is this test ? If so it should starts with 'test_'

TatyanaDovzhich replied:
Fixed:  test  was renamed into "test_sign_out_from_profile_page_after_sign_in"

QALightLecorPython commented on pages/header.py:
```diff
@@ -36,16 +39,37 @@ def sign_out_button(self, username, password):
+        assert not self.is_element_exists(self.const.MY_PROFILE_BUTTON_XPATH), "My Profile button didn't disappear"
+
+    @log_wrapper
+    def my_profile_button(self, username, password):
```
Do you really need such validation for this button ?

TatyanaDovzhich replied:
Fixed: validation method for My Profile button was deleted

QALightLecorPython reviewed (changes requested):
Looks good, but I have a few comments

TatyanaDovzhich committed:
[Lesson 17_Homework_17] :
- renamed test to "test_..." into test_hello_page.py & test_profile_page.py
- deleted strings with "return profile_page" into test_hello_page.py & test_profile_page.py
- validation method for My Profile button was deleted into pages/header.py
